### PR TITLE
fix err in keyword searching

### LINKED ISSUE #470: Search error: failed to fetch count: pq: syntax error at or near "AGAINST" (500)
Version: 
ghcr.io/go-shiori/shiori:v1.5.3-6-g3deeda7
postgres:14.5-alpine3.16

log:
```
time="2022-08-17T05:14:11Z" level=warning msg="GET /api/bookmarks?keyword=ndt&tags=&exclude=&page=1" proto=HTTP/1.1 remote="10.42.1.25:45580" reqlen=0 size=61 status=500
```
```
2022-08-17 05:14:11.273 UTC [7045] ERROR:  syntax error at or near "AGAINST" at character 101
2022-08-17 05:14:11.273 UTC [7045] STATEMENT:  SELECT COUNT(id) FROM bookmark WHERE TRUE AND (
				url LIKE $1 OR
				MATCH(title, excerpt, content) AGAINST ($2 IN BOOLEAN MODE)
			)
```

### LINKED ISSUE #477: Error when searching for a keyword that does not exist in the database
Just try typing in the search field something like "asdjalksdjlasjd".

> failed to expand bookmark_tag query: empty slice passed to 'in' query (500)

I use `sqlite` backend. Apparently it happens here:

https://github.com/go-shiori/shiori/blob/3deeda7151b8d719df2d9b7ac74cdcad16aa805e/internal/database/sqlite.go#L351-L359

And apparently the `bookmarkIds` array is empty.

Shiori v1.5.3.

### PR DESCRIPTION
**Pull request by hulb**
fix #470 #477

### CODE REVIEW

**Comment by hulb** [comment]
fix #470 #477

File: internal/database/pg.go [hunk 1]
```diff
@@ -223,7 +223,9 @@ func (db *PGDatabase) GetBookmarks(ctx context.Context, opts GetBookmarksOptions
 	if opts.Keyword != "" {
 		query += ` AND (
 			url LIKE :lkw OR
-			MATCH(title, excerpt, content) AGAINST (:kw IN BOOLEAN MODE)
```
Comment (hulb) [inline internal/database/pg.go:226]: It seems that MATCH ... AGAINST is a mysql syntax for text searching. In postgres, I don't find such simple way to implement text matching so I use LIKE multi times as a workround.

Comment (fmartingr) [reply internal/database/pg.go:226]: Yeah, we need to implement FTS into PSQL, I need to take a look at #276 first in order to decide an engine.

**Comment by fmartingr** [comment]
Hey @hulb, as always, thank you for fixing this! Little by little we're making Shiori more stable and usable :)

**Pull request merged**